### DISCUSSION TITLE
support 'GRPC Server Reflection Protocol'

### DISCUSSION THREAD

**Discussion by hartmut-co-uk** [Kong/insomnia#3669] [Ideas] [answered]
**Is your feature request related to a problem? Please describe.**
As an alternative to adding directories or individual .proto file(s) from disk - I'd also like to use _GRPC Server Reflection Protocol_ if available to load+reload Proto schema.

**Describe the solution you'd like**
Load proto from server according to official protocol:
https://github.com/grpc/grpc/blob/master/doc/server-reflection.md

**Describe alternatives you've considered**

**Additional context**

**Comment by redhoyasa** [comment]
I believe this is something similar to https://github.com/uw-labs/bloomrpc/pull/314 right?

**Reply by farislr** [reply]
I didn't know bloomrpc has the ability to use a reflection server. I usually use wombat https://github.com/rogchap/wombat

**Comment by wdawson** [comment]
Copying these comments over from #3942 (a duplicate)

From @Bouwdie 
> Hello everyone. I am very happy with the coming of GRPC into the insomnia tooling. I think it is an opportunity to get ahead of the game in API development and testing - one tool to rule them all (whilst keeping it simple). One thing that might be an interesting feature is an implementation of the GRPC reflection protocol: https://github.com/grpc/grpc/blob/master/doc/server-reflection.md It would make connecting to services that much easier since you do not have to acquire the proto files. Which for a simple service is a nuisance but doable, for a more complex one using the buf toolset with external dependencies it gets more complex. Personally (and in my company) we implement the reflection protocol for all our services to lower the boundaries of starting with GRPC. I think it would not be difficult to add, and an easy step to make GRPC more accessible. What are your thoughts? 🔢

From @dimitropoulos 
> I'm very excited to hear ideas like this! Thanks!
> 
> The first thing I'd like to know is: could you assist in getting a testing setup going so we can make a server that we can test implementing this with?

**Comment by alvian-pratama** [comment]
adding to this discussion, I believe this feature also exist on Postman implementation of grpc

also, to enable the reflection on server, on Go it can simply be done by adding this line after registering the grpc server
```
	// register server (srv is the variable of new grpc server) using reflection
	reflection.Register(srv) 
```

**Comment by jackkav** [comment, marked as answer]
We're working this in #5518
